### DISCUSSION TITLE
How to migrate StreamedAsyncHandler in v3?

### DISCUSSION THREAD

**Discussion by amanteaux** [AsyncHttpClient/async-http-client#1925] [General]
I was using `StreamedAsyncHandler` in version 2.12.3 to get the response body as a `org.reactivestreams.Publisher`.

I tried to use the 3.0.0.Beta3 version and this interface is not present anymore.

Is there any plan to add it? Or else, do you have any idea of how I can migrate my code to be able to get the response body as a `org.reactivestreams.Publisher`?

**Comment by kciesielski** [comment]
sttp, a leading Scala http library would like to update to async-http-client 3.x, but we're dealing with the same issue (https://github.com/softwaremill/sttp/pull/2241)
Reactive Streams support has been removed from async-http-client 3 [here](https://github.com/AsyncHttpClient/async-http-client/commit/1b63e0b00d5cd9f2317d855341e0324d0b2b3f47#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L278)
@hyperxpro should we consider this integration no longer available, or is there a recommended migration?

**Reply by hyperxpro** [reply]
Yes, I no longer plan to maintain Reactive streams. 

There are 2 strong reasons behind it:
- Other libraries provide better reactive stream support. Like reactor-netty.
- Maintenance overhead
